src/celeritas/ext/VecgeomParams.cc

```diff
@@ -82,6 +83,40 @@ VecgeomParams::VecgeomParams(const std::string& filename)
         vgdml::Frontend::Load(filename, validate_xml_schema);
     }
 
+    Initialize();
+}
+
+#ifdef CELERITAS_USE_GEANT4
+//---------------------------------------------------------------------------//
+/*!
+ *  Create a VecGeom model from an pre-existing Geant4 geometry
+ */
+VecgeomParams::VecgeomParams(const G4VPhysicalVolume* p_G4world)
+{
+    CELER_LOG(info) << "Creating VecGeom model from pre-existing G4 geometry";
+    CELER_ASSERT(p_G4world);
+
+    // Convert the geometry to VecGeom
+    G4VecGeomConverter::Instance().SetVerbose(1);
+    G4VecGeomConverter::Instance().ConvertG4Geometry(p_G4world);
+    CELER_LOG(info) << "Converted: max_depth = "
+                    << vecgeom::GeoManager::Instance().getMaxDepth()
+                    << "\nTop G4 volume: " << p_G4world->GetName();
+
+    //.. dump VecGeom geometry details for comparison
+    vecgeom::VPlacedVolume const* vgWorld
+        = vecgeom::GeoManager::Instance().GetWorld();
+    CELER_ENSURE(vgWorld);
+    CELER_LOG(info) << "Top VecGeom volume: " << vgWorld->GetName()
+                    << " - Label: " << vgWorld->GetLabel();
+
+    Initialize();
+}
+#endif
+
+void VecgeomParams::Initialize()
+{
+    CELER_LOG(status) << "Checking VecGeom volume labels";
```

Comment:
Should be debug maybe or deleted... the volume label construction isn't going to be slow or crash-prone, which is the main reason to put a "status" I think.

Reply:
Also you might add
```c++
CELER_EXPECT(vecgeom::GeoManager::Instance().GetWorld());
```
at the top of the function.